mesonbuild/backend/backends.py

```diff
@@ -1598,22 +1598,23 @@ def eval_custom_target_command(
         cmd = [i.replace('\\', '/') for i in cmd]
         return inputs, outputs, cmd
 
+    def get_introspect_command(self) -> str:
+        return ' '.join([shlex.quote(x) for x in self.environment.get_build_command() + ['introspect']])
```

Comment:
join should accept a generator expression, no need for a list comprehension here.

Reply:
I agree and it is also how it is implemented in `shlex.join` (I thought that maybe there was some obscure size-hint allocation benefit from using a list). Yet, I chose to use the pattern that is used everywhere else in `meson` for consistency.